vmdb/app/models/provider_foreman.rb

```diff
@@ -0,0 +1,18 @@
+class ProviderForeman < Provider
+  has_one :configuration_manager, :class_name => "ConfigurationManagerForeman", :dependent => :destroy
+  has_one :provisioning_manager,  :class_name => "ProvisioningManagerForeman",     :dependent => :destroy
+
+  def before_save
+    build_provisioning_manager unless provisioning_manager
+    build_configuration_manager unless configuration_manager
+  end
```

Comment:
Prefer using [inheritable callbacks vs override](http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html#module-ActiveRecord::Callbacks-label-Inheritable+callback+queues)

```ruby
  before_save :init_managers

  private

  def init_managers
    build_provisioning_manager unless provisioning_manager
    build_configuration_manager unless configuration_manager
  end
```

Also, should this be on before_create instead of before_save?  The latter will be called much more frequently.